hack/check_upgrade_ovs.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2020 Red Hat, Inc.
+#
+
+set -ex
+
+if [[ ${RELEASE_DELTA} == 2 ]]; then
+  # upgrading from the previous minor version
+  if [[ -z ${PREVIOUS_OVS_ANNOTATION} ]] || [[ ${PREVIOUS_OVS_ANNOTATION} == 'true' ]]; then
+    # if the annotation is "true" or not existed in previous version - OVS should be deployed in the new version.
+    echo "check that OVS annotation in HCO CR in new version is set to true"
+    [[ $(${CMD} get HyperConverged kubevirt-hyperconverged -n kubevirt-hyperconverged -o jsonpath='{.metadata.annotations.deployOVS}') == 'true' ]]
+
+    echo "check that OVS exists in CNAO CR Spec"
+    [[ $(${CMD} get networkaddonsconfigs cluster -o jsonpath='{.spec.ovs}') == '{}' ]]
+
+    echo "check that OVS DaemonSet exists"
+    [[ $(${CMD} get ds ovs-cni-amd64 -n kubevirt-hyperconverged --no-headers --ignore-not-found | wc -l) == '1' ]]
+
+  elif [[ -n ${PREVIOUS_OVS_ANNOTATION} ]] && [[ ${PREVIOUS_OVS_ANNOTATION} != 'true' ]]; then
+    # if the annotation existed in the previous version and was not "true" - OVS should not be deployed.
+      echo "check that OVS does not exist in CNAO CR Spec"
+      [[ $(${CMD} get networkaddonsconfigs cluster -o jsonpath='{.spec.ovs}') == '' ]]
+
+      echo "check that OVS DaemonSet does not exist"
+      [[ $(${CMD} get ds ovs-cni-amd64 -n kubevirt-hyperconverged --no-headers --ignore-not-found | wc -l) == '0' ]]
+  fi
+
+elif [[ ${RELEASE_DELTA} == 1 ]]; then
```

Comment:
I don't understand why there is a difference if we are upgrading from previous minor version or from same version, can you elaborate?

Reply:
if we are upgrading from previous minor version, the annotation does not exist in the previous version, and upon upgrade to the current version (CNV 2.6), we'll have to deploy ovs and add the true annotation.
if we are upgrading from the same version, we already have this opt-in machanism, thus: annotation not exists == no OVS is deployed.
that's why we have a distinction between the two cases.

Reply:
this logic is different from what HCO actually does, which is to check if ovs was previously deployed (in absence of the annotation during upgrade) and add the annotation accordingly.
The distinction between the types of RELEASE_DELTA is not a close fit to what we do in the code logic, and might break. it might also do a disservice as documentation of what HCO actually does.

I suggest checking if OVS is deployed in CNAO CR before the upgrade (similar to how we pre-save the PREVIOUS_OVS_ANNOTATION) and act upon it here